__tests__/restoreOnly.test.ts

```diff
@@ -0,0 +1,322 @@
+import * as cache from "@actions/cache";
```

Comment:
Are these tests same as restore.tests.ts. If yes then they are duplicate. We should move these tests to restoreImpl.tests.ts and keep only very minimal tests in restore.tests.ts and restoreOnly.tests.ts. For example these tests can verify that output is set in case of restoreOnly instead of state and vice versa.